main.go

```diff
@@ -13,122 +13,143 @@
 package main
 
 import (
+	"bytes"
+	"context"
+	"errors"
 	"flag"
 	"fmt"
 	"io/ioutil"
 	"os"
 	"os/exec"
+	"os/signal"
 	"regexp"
 	"runtime"
-	"sync/atomic"
-	"syscall"
+	"sync"
 	"time"
 )
 
 var (
-	flagP        = flag.Int("p", runtime.NumCPU(), "run `N` processes in parallel")
-	flagTimeout  = flag.Duration("timeout", 0, "timeout each process after `duration`")
-	flagKill     = flag.Bool("kill", true, "kill timed out processes if true, otherwise just print pid (to attach with gdb)")
-	flagFailure  = flag.String("failure", "", "fail only if output matches `regexp`")
-	flagIgnore   = flag.String("ignore", "", "ignore failure if output matches `regexp`")
-	flagMaxTime  = flag.Duration("maxtime", 0, "maximum time to run")
-	flagMaxRuns  = flag.Int("maxruns", 0, "maximum number of runs")
-	flagMaxFails = flag.Int("maxfails", 1, "maximum number of failures")
-	flagStdErr   = flag.Bool("stderr", true, "output failures to STDERR instead of to a temp file")
+	flags        = flag.NewFlagSet(os.Args[0], flag.ContinueOnError)
+	flagP        = flags.Int("p", runtime.NumCPU(), "run `N` processes in parallel")
+	flagTimeout  = flags.Duration("timeout", 0, "timeout each process after `duration`")
+	flagKill     = flags.Bool("kill", true, "kill timed out processes if true, otherwise just print pid (to attach with gdb)")
+	flagFailure  = flags.String("failure", "", "fail only if output matches `regexp`")
+	flagIgnore   = flags.String("ignore", "", "ignore failure if output matches `regexp`")
+	flagMaxTime  = flags.Duration("maxtime", 0, "maximum time to run")
+	flagMaxRuns  = flags.Int("maxruns", 0, "maximum number of runs")
+	flagMaxFails = flags.Int("maxfails", 1, "maximum number of failures")
+	flagStdErr   = flags.Bool("stderr", true, "output failures to STDERR instead of to a temp file")
```

Comment:
This *really* should be the only API. I hate the global mutable flags state. /rant